File: serveur/app/models/actual_workout_set.py
# Exemple : app/models/actual_workout_set.py

from sqlalchemy import Column, Integer, Float, Boolean, ForeignKey, String, DateTime, func
from app.database import Base

import uuid

class ActualWorkoutSet(Base):
    __tablename__ = "actual_workout_sets"

    id = Column(Integer, primary_key=True, index=True)
    actual_workout_exercise_uuid = Column(String, ForeignKey("actual_workout_exercises.uuid", ondelete="CASCADE"), nullable=False, index=True)
    set_order = Column(Integer, nullable=False)
    reps = Column(Integer, nullable=False)
    weight = Column(Float, nullable=False)
    is_dropset = Column(Boolean, nullable=False, default=False, server_default="false")
    notes = Column(String, nullable=True)
    recommendation = Column(String, nullable=True)
    status = Column(String, nullable=False, default="NOT_STARTED", server_default="NOT_STARTED")

    updated_at = Column(DateTime(timezone=True), nullable=True)

    uuid = Column(String, default=lambda: str(uuid.uuid4()), nullable=False, unique=True)

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/ActualWorkoutSet.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.ForeignKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import java.util.UUID

@Entity(
    tableName = "actual_workout_sets",
    indices = [
        androidx.room.Index(value = ["uuid"], unique = true),
        androidx.room.Index(value = ["actual_workout_exercise_uuid"]),
    ],
    foreignKeys = [
        ForeignKey(
            entity = ActualWorkoutExercise::class,
            parentColumns = ["uuid"],
            childColumns = ["actual_workout_exercise_uuid"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class ActualWorkoutSet(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "actual_workout_exercise_uuid") val actualWorkoutExerciseUUID: String,
    @ColumnInfo(name = "set_order") val setOrder: Int = 0,
    val reps: Int,
    val weight: Float,
    @ColumnInfo(name = "is_dropset") val isDropset: Boolean = false,
    val notes: String? = null,
    val recommendation: String? = null,
    val status: String,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
